src/main/java/org/testng/internal/TestInvoker.java

```diff
@@ -207,7 +208,13 @@ public FailureContext retryFailed(
   }
 
   public void runTestResultListener(ITestResult tr) {
-    TestListenerHelper.runTestListeners(tr, m_notifier.getTestListeners());
+    if(tr.getStatus() == ITestResult.STARTED) {
+      TestListenerHelper.runTestListeners(tr, m_notifier.getTestListeners());
+    } else {
+      List<ITestListener> test_Listeners_reverted = Lists.newArrayList(m_notifier.getTestListeners());
```

Comment:
Please have this variable name renamed to `testListenersReversed` instead of `test_Listeners_reverted`

Reply:
Done